src/Homework6/ReverseLinkedList.java

```diff
@@ -0,0 +1,28 @@
+package Homework6;
+
+import java.util.LinkedList;
+
+public class ReverseLinkedList {
+    public static void main(String[] args) {
+        LinkedList<Integer> list = new LinkedList<>();
+        list.add(1);
+        list.add(2);
+        list.add(3);
+        list.add(4);
+        list.add(5);
+
+        System.out.println("Original List: " + list);
+
+        LinkedList<Integer> reversedList = reverseLinkedList(list);
+
+        System.out.println("Reversed List: " + reversedList);
+    }
+
+    public static LinkedList<Integer> reverseLinkedList(LinkedList<Integer> list) {
+        LinkedList<Integer> reversedList = new LinkedList<>();
+        for (int i = list.size() - 1; i >= 0; i--) {
+            reversedList.add(list.get(i));
```

Comment:
it can work work fine for array list, but for linked list it takes O(n) for getting element by the index. Consider using links to previous and next elements to reverse the linked list